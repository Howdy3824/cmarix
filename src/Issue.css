.issue {
  border: 1px solid #ddd;
  display: flex;
  padding: 0.5rem;
}

.issue__number {
  font-size: 14px;
  color: #999;
  margin-right: 0.5rem;
}

.issue__title {
  font-weight: bold;
}

.issue__summary {
  
}

.issue__user {
  text-align: center;
  margin-right: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.issue__user__avatar {
  width: 64px;
  height: 64px;
}

.issue__user__name {
  color: #555;
  font-size: 0.8rem;
}

.issue__label {
  font-size: 0.9rem;
  padding: 3px 5px;
  margin-right: 3px;
  border-radius: 5px;
  border: 2px solid #ccc;
}