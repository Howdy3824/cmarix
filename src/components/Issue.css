.issue {
  border-bottom: 1px solid #ddd;
  display: flex;
  padding: 1rem 0.5rem;
}

.issue__number {
  font-size: 14px;
  color: #999;
  margin-right: 0.5rem;
}

.issue__title {
  font-weight: bold;
}

.issue__label {
  display: inline-block;
  font-size: 0.75rem;
  padding: 1px 5px;
  margin: 0 0.5rem 0.5rem 0;
  border-radius: 5px;
  border: 1px solid #ccc;
  background-color: #fff;
}

.issue .issue__user {
  margin-right: 0.5rem;
}

.issue__summary {
  padding-left: 0.5rem;
}

.issue__labels {
  padding-left: 0.5rem;
}

.issue__body {
  word-wrap: break-word;
  overflow: hidden;
}

.issue--loading .issue__body {
  width: 100%;
}
.issue--loading .issue__number,
.issue--loading .issue__title {
  background-color: #e5e5e5;
  display: inline-block;
  margin-right: 0.5rem;
}
.issue--loading .issue__number {
  width: 20%;
  max-width: 4rem;
}
.issue--loading .issue__title {
  width: 70%;
}
.issue--loading .issue__summary {
  background-color: #e5e5e5;
  width: 95%;
}