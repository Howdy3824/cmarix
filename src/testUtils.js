import toJson from 'enzyme-to-json';

export function afterPromises(done, fn) {
  setTimeout(() => {
    try {
      fn();
      done();
    } catch (e) {
      fail();
    }
  }, 1);
}

export function checkSnapshot(tree) {
  expect(toJson(tree)).toMatchSnapshot();
}