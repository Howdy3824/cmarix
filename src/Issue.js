import React, { PropTypes } from 'react';
import './Issue.css';

function shorten(text = "", length = 140) {
  // Normalize newlines
  let cleanText = text.replace(/\\r\\n/g, "\\n");

  // Return if short enough already
  if(cleanText.length <= length) {
    return cleanText;
  }

  return cleanText.substr(0, 140);
}

export default function Issue({ number, title, labels, username, avatarUrl, summary }) {
  return (
    <div className="issue">
      <div className="issue__user">
        <img className="issue__user__avatar" src={avatarUrl} alt=""/>
        <div className="issue__user__name">{username}</div>
      </div>
      <div>
        <div>
          <span className="issue__number">#{number}</span>
          <span className="issue__title">{title}</span>
        </div>
        <p className="issue__summary">{shorten(summary)}</p>
        <div className="issue__labels">
          {labels.map(label =>
            <span className="issue__label" key={label}>
              {label}
            </span>
          )}
        </div>
      </div>
    </div>
  );
}

Issue.propTypes = {
  number: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
  username: PropTypes.string.isRequired,
  avatarUrl: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired
};