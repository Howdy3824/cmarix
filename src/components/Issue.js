import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import UserWithAvatar from './UserWithAvatar';
import IssueLabels from './IssueLabels';
import { shorten } from '../utils/stringUtils';
import './Issue.css';

export default function Issue({ number, title, labels, user, summary }, { router }) {
  const {org, repo} = router.params;
  return (
    <div className="issue">
      <UserWithAvatar user={user}/>
      <div className="issue__body">
        <Link to={`/${org}/${repo}/issues/${number}`}>
          <span className="issue__number">#{number}</span>
          <span className="issue__title">{title}</span>
        </Link>
        <p className="issue__summary">{shorten(summary)}</p>
        <IssueLabels labels={labels}/>
      </div>
    </div>
  );
}

Issue.propTypes = {
  number: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  labels: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    color: PropTypes.string
  })).isRequired,
  user: PropTypes.shape({
    login: PropTypes.string.isRequired,
    avatar_url: PropTypes.string
  }).isRequired,
  summary: PropTypes.string.isRequired
};

Issue.contextTypes = {
  router: PropTypes.object.isRequired
};