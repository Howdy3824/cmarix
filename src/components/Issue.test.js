import React from 'react';
import { shallow } from 'enzyme';
import { checkSnapshot } from '../utils/testUtils';
import Issue, { shorten } from './Issue';

it('renders', () => {
  const context = { router: {params: {org: 'rails', repo: 'rails'}} };
  const tree = shallow(
    <Issue
        number={27593}
        title="Yield in partial view does not call block given by calling view"
        user={{
          login: "somebody",
          avatar_url: "https://avatars.githubusercontent.com/u/2460418?v=3"
        }}
        labels={[{
          id: 1, name: "activerecord", color: "123456"
        }]}
        summary="Fixes #23979.\r\n\r\nAs discussed in #23979, there was an inconsistency between the way that `first()` and `last()` would interact with `limit`. Specifically:\r\n\r\n```Ruby\r\n> Topic.limit(1).first(2).size\r\n=> 2\r\n> Topic.limit(1).last(2).size\r\n=> 1\r\n```\r\n\r\nThis PR is a refactor and rebase of #24124, with a simpler test suite and simpler implementation.\r\n\r\nDiscussion with Rails community members as well as DHH in https://github.com/rails/rails/pull/23598#issuecomment-189675440 showed that the behavior or `first` should be brought into line with `last` (rather than vice-versa).\r\n\r\nThis PR resolves the inconsistency between `first` and `last` when used in conjunction with `limit`.\r\n"
      />, { context }
  );

  checkSnapshot(tree);
});