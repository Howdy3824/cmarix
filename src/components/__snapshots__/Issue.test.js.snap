exports[`test renders 1`] = `
<div
  className="issue">
  <UserWithAvatar
    user={
      Object {
        "avatar_url": "https://avatars.githubusercontent.com/u/2460418?v=3",
        "login": "somebody",
      }
    } />
  <div
    className="issue__body">
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/rails/rails/issues/27593">
      <span
        className="issue__number">
        #
        27593
      </span>
      <span
        className="issue__title">
        Yield in partial view does not call block given by calling view
      </span>
    </Link>
    <p
      className="issue__summary">
      Fixes #23979.
      
      As discussed in #23979, there was an inconsistency between the way that \`first()\` and \`last()\` would interact with \`limit\`. ...
    </p>
    <IssueLabels
      labels={
        Array [
          Object {
            "color": "123456",
            "id": 1,
            "name": "activerecord",
          },
        ]
      } />
  </div>
</div>
`;
